detox/src/devices/DeviceRegistry.js

```diff
@@ -45,7 +45,7 @@ class DeviceRegistry {
   }
 
   includes(deviceId) {
-    return this._lockfile.read().includes(deviceId);
+    return !!_.find(this._lockfile.read(), (item) => _.isEqual(item, deviceId));
```

Comment:
I would not hurry to agree with your approach here, Amit.
I'd rather use small wrapper classes where you define your own `equals(other: T): boolean` logic.
WDYT?

Reply:
See 1: https://github.com/wix/Detox/pull/2466#discussion_r529422120
See 2: https://github.com/wix/Detox/pull/2466#discussion_r529404752
See 3: https://github.com/wix/Detox/pull/2466#discussion_r529420727
See 4: https://github.com/wix/Detox/pull/2466#discussion_r529431175

Reply:
The funny thing is that this is how I would undoubtedly have done this in Java/Kotlin. Seems I've underestimated JS. I will definitely try to do this as you suggested 💪🏻

Reply:
So I haven't anything idiomatic that corresponds to your suggestion (namely such that evidently, an equality operator would result in calling a custom function). Could you pinpoint me here?